### PR TITLE
httpserver: support QUIC reload

### PR DESCRIPTION
**Pull request by tw4452852**
fix issue #958

Signed-off-by: Tw <tw19881113@gmail.com>

### CODE REVIEW

File: caddyhttp/httpserver/server.go [hunk 1]
```diff
@@ -146,7 +146,16 @@ func (s *Server) Listen() (net.Listener, error) {
 }
 
 // ListenPacket is a noop to implement the Server interface.
```
Comment (mholt) [inline caddyhttp/httpserver/server.go]: No longer a no-op. ;) Need to update the godoc comment.

File: caddyhttp/httpserver/server.go [hunk 2]
```diff
@@ -189,7 +189,12 @@ func (s *Server) Serve(ln net.Listener) error {
 }
 
 // ServePacket is a noop to implement the Server interface.
-func (s *Server) ServePacket(pc net.PacketConn) error { return nil }
+func (s *Server) ServePacket(pc net.PacketConn) error {
+	if QUIC {
+		return s.quicServer.Serve(pc.(*net.UDPConn))
```
Comment (mholt) [inline caddyhttp/httpserver/server.go]: Could we wrap this error, something like what we used to have:

```go
return fmt.Errorf("listening for QUIC connections: %v", err)
```

File: caddyhttp/httpserver/server.go (hunk 1 above)
Comment (tw4452852) [inline caddyhttp/httpserver/server.go]: Done.

File: caddyhttp/httpserver/server.go (hunk 2 above)
Comment (tw4452852) [inline caddyhttp/httpserver/server.go]: Done

**Comment by tw4452852** [comment]
Yes, I try to reload the server with `SIGUSR1`, it works well.

**Comment by mholt** [comment]
Thank you! This is awesome.

**Pull request merged**